tests/api/products.js

```diff
@@ -0,0 +1,179 @@
+import path from "node:path";
+import fs from "node:fs/promises";
+import { globSync } from "glob";
+
+const PRODUCT_TYPE_IN_PATH_RX = /\/products\/types\/([A-Za-z]+)(\/|\.json)/;
+const PRODUCT_TYPE_LOCATION_RX =
+  /\/products\/types\/[A-Za-z]+\/locations\/([A-Za-z]+)\.json/;
+const PRODUCT_TYPE_ALL_RX = /\/products\/types\/[A-Za-z]+\.json/;
+const PRODUCT_TYPE_INDIVIDUAL_RX = /\/products\/[^/]+\.json/;
+
+/**
+ * @param {string} base - The base dir for bundles
+ * @param {string} bundleName - The name of the bundle
+ * @returns {string[]} - An array of full file paths for any matching
+ *            JSON files under the products for the given bundle
+ */
+const getProductFilePaths = (base, bundleName) => 
+      globSync(path.join(base, bundleName, "products", "**", "*.json"));
+
+/**
+ * Attempts to extract the product code from the path
+ * @param {string} filePath - The path to the given JSON file
+ * @returns {string?} - The extracted product type code or null
+ */
+const getProductTypeFromPath = (filePath) => {
+  const match = filePath.match(PRODUCT_TYPE_IN_PATH_RX);
+  if (!match) {
+    return null;
+  }
+
+  return match[1];
+};
+
+/**
+ * Attempts to extract the product code from
+ * the parsed JSON data object for this product.
+ * @param {object} data - The parsed JSON object for this
+ * individual product
+ * @returns {string} - The extracted product code or UNKNOWN
+ */
+const getProductTypeFromData = (data) => 
+      data.productCode || "UNKNOWN";
+
+/**
+ * Attempts to extract a WFO code from the filePath
+ * @param {string} filePath - A path to a JSON file
+ * @returns {string?} - The extracted WFO code, if found,
+ * or null otherwise
+ */
+const getWFOCodeFromPath = (filePath) => {
+  const match = filePath.match(PRODUCT_TYPE_LOCATION_RX);
+  if (!match) {
+    return null;
+  }
+
+  return match[1];
+};
+
+/**
+ * Composes a label for the bundled product file that
+ * will be used for displaying information about the bundler
+ * in its UI.
+ * @param {object} data - The parsed JSON data object
+ * @param {string} filePath - The path to the source JSON file
+ * @param {string} recordType - 'individual', 'all', or 'location'
+ * @returns {string} - A label that describes this data file for the
+ * bundler UI
+ */
+const getProductLabel = (data, filePath, recordType) => {
+  if (data["@bundle"]?.name) {
+    return data["@bundle"].name;
+  }
+  const productType =
+    getProductTypeFromPath(filePath) ?? getProductTypeFromData(data);
+  const wfoCode = getWFOCodeFromPath(filePath);
+  let size;
+  if (recordType === "all" || recordType === "location") {
+    size = data["@graph"].length;
+  }
+  if (recordType === "all") {
+    return `List of most recent ${productType}s (sample of ${size})`;
+  }
+  if (recordType === "location") {
+    return `List of most recent ${productType}s for office ${wfoCode} (sample of ${size})`;
+  }
```

Comment:
Suggestion in the spirit of limiting scope pollution:

```suggestion
  if (recordType === "all") {
    const size = data["@graph"].length;
    return `List of most recent ${productType}s (sample of ${size})`;
  }
  if (recordType === "location") {
    const size = data["@graph"].length;
    return `List of most recent ${productType}s for office ${wfoCode} (sample of ${size})`;
  }
}
```